addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -1048,6 +1046,14 @@ odoo.define('pos_coupon.pos', function (require) {
                 this.is_program_reward = json.is_program_reward;
                 this.program_id = json.program_id;
                 this.coupon_id = json.coupon_id;
+                if (json.program_id && json.program_id[0]) {
```

Comment:
A promotion program does not have a coupon_id 
But a coupon has a program_id assigned so with the current code also coupon end up in `activePromoProgramIds`